### DISCUSSION TITLE
Would like Caffeine instance builder to support an objectServedFromCacheListener

### DISCUSSION THREAD

**Discussion by yschondorf-cr** [ben-manes/caffeine#555] [Ideas]
When creating a Caffeine bean the builder is used as in the following example:

@Bean
public Caffeine caffeineConfig() {
    return Caffeine.newBuilder().maximumSize(1000).expireAfterWrite(4, TimeUnit.HOURS);
}
The builder has a removalListener() to attach a listener for removal events. I wish there was such an easy way to attach a listener for other events such as cache hits and misses. My particular use case needed a objectServedFromCacheListener, but I can see this request easily extended to support objectNotInCacheListener and objectSavedInCacheListener.

**Comment by ben-manes** [comment]
The primary use-case for this is for statistics, which you can enable on the builder. That can be given its own `StatsCounter` to record with if you want to push them, rather than poll via `cache.stats()`. This does not capture key/value fields, only the metrics.

Otherwise you can manually intercept via a `ThreadLocal` that is set within the `CacheLoader` if invoked. This way the caller can check and clear the flag to determine if a load occurred. If instead you use `cache.get(key, mappingFunction)`, then the lamda can capture a local variable and set the flag before returning the value. Either approach lets you know if the loading function was used, and react accordingly
